jina/serve/runtimes/gateway/http/app.py

```diff
@@ -227,6 +228,72 @@ async def _render_custom_swagger_html(req: Request) -> HTMLResponse:
 
         app.add_route(docs_url, _render_custom_swagger_html, include_in_schema=False)
 
+    if not args.no_graphql_endpoint:
+        with ImportExtensions(required=True):
+            from dataclasses import asdict
+
+            import strawberry
+            from docarray import DocumentArray
+            from strawberry.fastapi import GraphQLRouter
+            from docarray.document.strawberry_type import StrawberryDocument
+            from docarray.document.strawberry_type import (
+                JSONScalar,
+                StrawberryDocumentInput,
+            )
+
+            async def get_docs_from_endpoint(
+                data, target_executor, parameters, exec_endpoint
+            ):
+                req_generator_input = {
+                    'data': [asdict(d) for d in data],
+                    'target_executor': target_executor,
+                    'parameters': parameters,
+                    'exec_endpoint': exec_endpoint,
+                    'data_type': DataInputType.DICT,
+                }
+
+                if (
+                    req_generator_input['data'] is not None
+                    and 'docs' in req_generator_input['data']
+                ):
+                    req_generator_input['data'] = req_generator_input['data']['docs']
+
+                response = await _get_singleton_result(
+                    request_generator(**req_generator_input)
+                )
+                return DocumentArray.from_dict(response['data']).to_strawberry_type()
+
+            @strawberry.type
+            class Mutation:
+                @strawberry.mutation
+                async def docs(
+                    self,
+                    data: Optional[List[StrawberryDocumentInput]] = None,
+                    target_executor: Optional[str] = None,
+                    parameters: Optional[JSONScalar] = None,
+                    exec_endpoint: str = '/search',
+                ) -> List[StrawberryDocument]:
+                    return await get_docs_from_endpoint(
+                        data, target_executor, parameters, exec_endpoint
+                    )
+
+            @strawberry.type
+            class Query:
```

Comment:
Why do we still need a `Query`?

Reply:
With strawberry you always have to have a Query, othws it doesn't work. Not sure if that's just strawberry or GraphQL in general, but we're stuck with it either way. So now query is not mentioned anywhere in the docs, but if someone tries it just does the same thing as mutation.